docs/jwst/nsclean/arguments.rst

```diff
@@ -6,19 +6,65 @@ Step Arguments
+  Ignored for subarray data and for `fit_method` = 'fft'.
+
+``--background_method`` (str, default=None)
+  If 'median', the preliminary background to remove and restore
+  after fitting the noise is a simple median of the background data.
+  If 'model', the background data is fit with a low-resolution model
+  via `~photutils.background.Background2D`.
+  If None, the background value is set to 0.0.
+
+``--background_box_size`` (list of int, default=(32,32))
+  Box size for the data grid used by `~photutils.background.Background2D`
+  when `background_method` = 'model'. For best results, use a
+  box size that evenly divides the input image shape.
+
 ``--mask_spectral_regions`` (boolean, default=True)
-  Mask regions in IFU and MOS images that are within the bounding boxes
-  for each slice or slitlet defined in the WCS object of the image.
+  Mask regions of the image defined by WCS bounding
+  boxes for slits/slices, as well as any regions known to be
+  affected by failed-open MSA shutters.
 
 ``--n_sigma`` (float, default=5.0)
   The sigma-clipping threshold to use when searching for outliers
-  and illuminated pixels to be excluded from use in the fitting
-  process.
+  and illuminated pixels to be excluded from use in the background
+  and noise fitting processes.
 
-``--save_mask`` (boolean, default=False)
-  A flag to indicate whether the mask constructed by the step
-  should be saved to a file.
+``--fit_histogram`` (boolean, default=False)
+  If set, the 'sigma' used with `n_sigma` for clipping outliers
+  is derived from a Gaussian fit to a histogram of values.
+  Otherwise, a simple iterative sigma clipping is performed.
+
+``--single_mask`` (boolean, default=False)
+  If set, a single mask will be created, regardless of
+  the number of input integrations. Otherwise, the mask will
+  be a 3D cube, with one plane for each integration.
 
 ``--user_mask`` (string, default=None)
   Path to a user-supplied mask file. If supplied, the mask is used
-  directly and the process of creating a mask in the step is skipped.
+  directly and the process of creating a scene mask in the step is
+  skipped.
+
+  The mask file must contain either a `~jwst.datamodels.ImageModel`
+  or a `~jwst.datamodels.CubeModel`, with image dimensions matching
+  the input science data.  If an ImageModel is provided, the same
+  mask will be used for all integrations.  If a CubeModel is provided,
+  the number of slices must equal the number of integrations in
+  the input science data.
+
+``--save_mask`` (boolean, default=False)
+  If set, the mask constructed by the step will be saved to a file
+  with suffix 'mask'.
+
+``--save_background`` (boolean, default=False)
+  If set, the background fit to the group diff images will be saved
+  to a file with suffix 'flicker_bkg'.
+
+``--save_noise`` (boolean, default=False)
+  If set, the residual noise fit and removed from the input data
+  will be saved to a file with suffix 'flicker_noise'.
```

Comment:
It's very minor, but I noticed when working on JP-3667 that the nsclean step is currently saving out the background and noise files as with the suffixes "background" and "noise" instead of "flicker_bkg" and 'flicker_noise' as listed here. It's not hugely critical, but I figured I'd mention it.

Reply:
Thanks for catching that!  It's fixed now.